GitVersionCore/Configuration/Config.cs

```diff
@@ -1,30 +1,90 @@
+
         public Config()
         {
             AssemblyVersioningScheme = AssemblyVersioningScheme.MajorMinorPatch;
-            DevelopBranchTag = "unstable";
-            ReleaseBranchTag = "beta";
             TagPrefix = "[vV]";
+            VersioningMode = GitVersion.VersioningMode.ContinuousDelivery;
+            ContinuousDeploymentFallbackTag = "ci";
+
+            Branches["master"] = new BranchConfig
+            {
+                Tag = string.Empty,
+                Increment = IncrementStrategy.Patch,
+                PreventIncrementOfMergedBranchVersion = true
+            };
+            Branches["release[/-]"] = new BranchConfig { Tag = "beta" };
+            Branches["feature[/-]"] = new BranchConfig
+            {
+                Increment = IncrementStrategy.Inherit,
+                Tag = "useBranchName"
+            };
+            Branches["hotfix[/-]"] = new BranchConfig { Tag = "beta" };
+            Branches["support[/-]"] = new BranchConfig
+            {
+                Tag = string.Empty,
+                Increment = IncrementStrategy.Patch,
+                PreventIncrementOfMergedBranchVersion = true
+            };
+            Branches["develop"] = new BranchConfig
+            {
+                Tag = "unstable",
+                Increment = IncrementStrategy.Minor,
+                VersioningMode = GitVersion.VersioningMode.ContinuousDeployment
+            };
+            Branches[@"(pull|pull\-requests|pr)[/-]"] = new BranchConfig
+            {
+                Tag = "PullRequest",
+                TagNumberPattern = @"[/-](?<number>\d+)[-/]",
+                Increment = IncrementStrategy.Inherit
+            };
         }
 
-        [YamlAlias("assembly-versioning-scheme")]
+        [YamlMember(Alias = "assembly-versioning-scheme")]
         public AssemblyVersioningScheme AssemblyVersioningScheme { get; set; }
 
-        [YamlAlias("develop-branch-tag")]
-        public string DevelopBranchTag { get; set; }
-
-        [YamlAlias("release-branch-tag")]
-        public string ReleaseBranchTag { get; set; }
+        [YamlMember(Alias = "mode")]
+        public VersioningMode? VersioningMode { get; set; }
 
-        [YamlAlias("tag-prefix")]
+        [YamlMember(Alias = "tag-prefix")]
         public string TagPrefix { get; set; }
 
-        [YamlAlias("next-version")]
+        [YamlMember(Alias = "continuous-delivery-fallback-tag")]
+        public string ContinuousDeploymentFallbackTag { get; set; }
+
+        [YamlMember(Alias = "next-version")]
         public string NextVersion { get; set; }
+
+        [YamlMember(Alias = "branches")]
+        public Dictionary<string, BranchConfig> Branches
+        {
+            get
+            {
+                return branches;
+            }
+            set
+            {
+                value.ToList().ForEach(_ => branches[_.Key] = MergeObjects(branches[_.Key],  _.Value));
```

Comment:
If one specify a new branch pattern (like 'development'), this `branches[_.Key]`will throw an exception, since it doesn't exist yet.